extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -74,6 +75,39 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 			}
 		} catch (e) {
 			vscode.window.showErrorMessage(loc.openFileError(bookPath, e instanceof Error ? e.message : e));
+		} finally {
+			// add file watcher on toc file.
+			fsw.watchFile(path.join(bookPath, '_data', 'toc.yml'), async (curr, prev) => {
```

Comment:
Is there a specific reason you're using watchFile? Seems like the suggestion is to use watch instead whenever possible https://nodejs.org/docs/latest/api/fs.html#fs_caveats